client/src/components/AuthProvider.jsx

```diff
@@ -0,0 +1,47 @@
+import { useState } from "react";
+import { AuthContext } from "../hook/auth.js";
+import { Auth } from "../services/authService.js";
+// extract email from jwt
+const decodeToken = (token) => {
+
+    let email = null;
+    if (token && token !== "undefined") {
+        const tokenPayload = token.split(".")[1];
+        const decodedPayload = atob(tokenPayload);
+        const parsedPayload = JSON.parse(decodedPayload);
+        email = parsedPayload.email;
+    }
+    return email
+}
+export const AuthProvider = ({ children }) => {
+    const token = sessionStorage.getItem("token");
+    const [state, setState] = useState({
+        token,
+        email: decodeToken(token),
+        error: null,
+    });
+    const login = async (user, password) => {
+        const res = await Auth.login(user, password);
+        console.log('login', { res });
+        if (res.err) {
+            console.error(res.err);
+            setState({ error: res.err, token: null });
+            return { error: res.err };
+        }
+        setState(({ error: null, token: res.token, email: decodeToken(res.token) }));
+        sessionStorage.setItem("token", res.token);
```

Comment:
Šiaip čia viskas gerai, bet manau, kad dėstytojas yra minęs, kad "auth token" saugojimas į sessionStorage arba localStorage nėra geriausia idėja. Tokią informaciją rekomenduoja saugoti, kaip secret cookie. Aišku, kaip ir visada, yra keletas tam tikrų limitaciją, pvz. cookie dydžio limitas.

Apie tai galite daugiau paskaityti stackoverflow.com klausime:
https://stackoverflow.com/questions/54258233/do-i-have-to-store-tokens-in-cookies-or-localstorage-or-session

Arba dev.to blog įraše:
https://dev.to/cotter/localstorage-vs-cookies-all-you-need-to-know-about-storing-jwt-tokens-securely-in-the-front-end-15id